### DISCUSSION TITLE
For those with PWs not on the local network..

### DISCUSSION THREAD

**Discussion by dalloro** [jasonacox/Powerwall-Dashboard#14] [Q&A] [answered]
Possibly naïve question, I am curious to hear if this stack supports pulling data from the Tesla cloud APIs, or if it requires to have the PWs on the local network?

Asking this, because mine are only connected via cellular and have no easy way to connect to them directly, except maybe by using a dedicated machine with a WiFi card connecting directly to the Tesla gateway access point. 

Currently getting data from the Tesla app, but would love to self-host.

**Comment by jasonacox** [comment, marked as answer]
Hi @dalloro - This is a great question.  The Tesla cloud APIs are a bit more difficult to access programmatically.  There are some projects trying to make it easier (see https://tesla-api.timdorr.com/ and https://github.com/mlowijs/tesla_api). 

You are right, this Powerwall-Dashboard stack is designed for local access to the PW gateway endpoint.  Your "dedicated machine" approach could be a workaround.  However, I would recommend a hardwire ethernet cable approach if you can. That's what I did. Very stable and fairly easy to do.
